Yona/Yona/Others/SMSValidPasscodeLogin/Login/LoginViewController.swift

```diff
@@ -206,15 +204,17 @@ extension LoginViewController: CodeInputViewDelegate {
         let passcode = KeychainManager.sharedInstance.getPINCode()
         if code ==  passcode {
             authenticatedSuccessFully()
+            UserDefaults.standard.set(0, forKey: YonaConstants.nsUserDefaultsKeys.numberOfLoginAttempts) // resetting number of login attempts to 0
         } else {
             errorLabel.isHidden = false
             self.codeInputView.clear()
-            if loginAttempts == maxAttempts {
+            var loginAttempts:Int = UserDefaults.standard.integer(forKey: YonaConstants.nsUserDefaultsKeys.numberOfLoginAttempts)
+            if  loginAttempts == maxAttempts {
                 showMaxLoginAttemptsReachedScreen()
```

Comment:
Is this also setting the (persistent) state in such a way that the user now needs to request a new PIN? And is that procedure resetting the failed attempts to 0?

Reply:
Yes, It is also where user needs to request a new PIN. In last commit, method is resetting the failed attempts to 0.